modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/plugin/ArtifactPluginDescriptorFactory.java

```diff
@@ -84,31 +87,42 @@ public ArtifactPluginDescriptorFactory(DescriptorLoaderRepository descriptorLoad
   }
 
   @Override
-  public ArtifactPluginDescriptor create(File pluginFolder) throws ArtifactDescriptorCreateException {
-    ArtifactPluginDescriptor artifactPluginDescriptor;
-    final File mulePluginJsonFile = new File(pluginFolder, MULE_ARTIFACT_FOLDER + separator + MULE_PLUGIN_JSON);
-    if (mulePluginJsonFile.exists()) {
-      artifactPluginDescriptor = loadFromJsonDescriptor(pluginFolder, mulePluginJsonFile);
-    } else {
-      // TODO(fernandezlautaro): MULE-11092 once implemented, drop this.
-      artifactPluginDescriptor = loadFromPluginProperties(pluginFolder);
+  public ArtifactPluginDescriptor create(File pluginJarFile) throws ArtifactDescriptorCreateException {
+    try {
+      checkArgument(pluginJarFile.getName().endsWith(".jar") || pluginJarFile.getName().endsWith(".zip"),
+                    "provided file is not a plugin: " + pluginJarFile.getAbsolutePath());
+      Optional<Byte[]> jsonDescriptorContentOptional =
+          loadFileContentFrom(pluginJarFile, MULE_ARTIFACT_FOLDER + separator + MULE_PLUGIN_JSON);
+      return jsonDescriptorContentOptional
+          .map(jsonDescriptorContent -> loadFromJsonDescriptor(pluginJarFile, new String(toPrimitive(jsonDescriptorContent))))
+          .orElseGet(() -> {
+            Optional<Byte[]> pluginPropertiesFileContentOptional;
+            try {
+              pluginPropertiesFileContentOptional = loadFileContentFrom(pluginJarFile, PLUGIN_PROPERTIES);
+              return loadFromPluginProperties(pluginJarFile, pluginPropertiesFileContentOptional);
+            } catch (IOException e) {
+              throw new MuleRuntimeException(e);
+            }
+          });
+    } catch (ArtifactDescriptorCreateException e) {
```

Comment:
Why? If the excepiton thrown is already an ArtifactDescriptorCreateException instance I don't want to wrap it.